src/main/java/com/ozonehis/ozone_demo_data/util/SystemAvailabilityChecker.java

```diff
@@ -0,0 +1,122 @@
+                return true;
+            } else {
+                log.warn("OpenMRS server is not available. Status code: {}", status);
+                return false;
+            }
+        } catch (Exception e) {
+            log.warn("OpenMRS Server not ready: {}", e.getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * This method waits for OpenMRS server to be available. It retries 5 times with a delay of 5 seconds between each
+     * retry.
+     *
+     * @return true if OpenMRS server is available, false otherwise.
+     */
+    public boolean waitForOpenMRSAvailability() {
+        int attempts = 0;
+        while (!isOpenMRSAvailable() && attempts < openmrsConfig.getMaxRetries()) {
+            try {
+                sleep(openmrsConfig.getRetryDelayMillis());
+            } catch (InterruptedException e) {
+                log.error("Error while waiting for OpenMRS server to be available", e);
+                Thread.currentThread().interrupt();
+                return false;
+            }
+            log.info("Waiting for OpenMRS server to be available...");
+            attempts++;
+        }
+        return isOpenMRSAvailable();
+    }
+
+    /**
+     * This method checks if Keycloak server is available or not. It sends a GET request to Keycloak health endpoint.
+     *
+     * @return true if Keycloak server is available, false otherwise.
+     */
+    public boolean isKeycloakAvailable() {
+        try {
+            HttpStatusCode status = restTemplate
+                    .getForEntity(keycloakConfig.getServerUrl() + "/health/ready", String.class)
+                    .getStatusCode();
+            if (status.is2xxSuccessful()) {
+                log.info("Keycloak server is available");
+                return true;
+            } else {
+                log.warn("Keycloak server is not available. Status code: {}", status);
+                return false;
+            }
+        } catch (Exception e) {
+            log.warn("Keycloak Server not ready: {}", e.getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * This method waits for Keycloak server to be available. It retries 5 times with a delay of 5 seconds between each
+     * retry.
+     *
+     * @return true if Keycloak server is available, false otherwise.
+     */
+    public boolean waitForKeycloakAvailability() {
+        int attempts = 0;
+        while (!isKeycloakAvailable() && attempts < keycloakConfig.getMaxRetries()) {
+            try {
+                sleep(keycloakConfig.getRetryDelayMillis());
+            } catch (InterruptedException e) {
+                log.error("Error while waiting for Keycloak server to be available", e);
+                Thread.currentThread().interrupt();
+                return false;
+            }
+            log.info("Waiting for Keycloak server to be available...");
+            attempts++;
+        }
+        return isKeycloakAvailable();
+    }
+}
```

Comment:
Seems like we are repeating a lot of logic here. Maybe we should create generic functions here which accept parameters.